npm_and_yarn/lib/dependabot/npm_and_yarn/file_updater/npm_lockfile_updater.rb

```diff
@@ -153,28 +165,127 @@ def run_npm_updater(lockfile_name:, top_level_dependency_updates:, lockfile_cont
+        end
+
+        def run_npm_7_top_level_updater(lockfile_name:, top_level_dependency_updates:)
+          # - `--dry-run=false` the updater sets a global .npmrc with dry-run: true to
+          #   work around an issue in npm 6, we don't want that here
+          # - `--force` ignores checks for platform (os, cpu) and engines
+          # - `--ignore-scripts` disables prepare and prepack scripts which are run
+          #   when installing git dependencies
+          args = npm_top_level_updater_args(top_level_dependency_updates)
+          command = [
+            "npm",
+            "install",
+            *args,
+            "--force",
+            "--dry-run",
+            "false",
+            "--ignore-scripts",
+            "--package-lock-only"
+          ].join(" ")
+          SharedHelpers.run_shell_command(command)
+          { lockfile_name => File.read(lockfile_name) }
         end
 
         def run_npm_subdependency_updater(lockfile_name:, lockfile_content:)
           npm_version = Dependabot::NpmAndYarn::Helpers.npm_version(lockfile_content)
-          Dependabot.logger.info(npm_version)
 
-          SharedHelpers.run_helper_subprocess(
-            command: NativeHelpers.helper_path,
-            function: "npm6:updateSubdependency",
-            args: [Dir.pwd, lockfile_name, sub_dependencies.map(&:to_h)]
-          )
+          if npm_version == "npm7"
+            run_npm_7_subdependency_updater(lockfile_name: lockfile_name)
+          else
+            SharedHelpers.run_helper_subprocess(
+              command: NativeHelpers.helper_path,
+              function: "npm6:updateSubdependency",
+              args: [Dir.pwd, lockfile_name, sub_dependencies.map(&:to_h)]
+            )
+          end
+        end
+
+        def run_npm_7_subdependency_updater(lockfile_name:)
+          dependency_names = sub_dependencies.map(&:name)
+          # - `--dry-run=false` the updater sets a global .npmrc with dry-run: true to
+          #   work around an issue in npm 6, we don't want that here
+          # - `--force` ignores checks for platform (os, cpu) and engines
+          # - `--ignore-scripts` disables prepare and prepack scripts which are run
+          #   when installing git dependencies
+          command = [
+            "npm",
+            "update",
+            *dependency_names,
+            "--force",
+            "--dry-run",
+            "false",
+            "--ignore-scripts",
+            "--package-lock-only"
+          ].join(" ")
+          SharedHelpers.run_shell_command(command)
+          { lockfile_name => File.read(lockfile_name) }
+        end
+
+        # TODO: Update the npm 6 updater to use these args as we currently do
+        # the same in the js updater helper, we've kept it seperate for the npm
+        # 7 rollout
+        def npm_top_level_updater_args(dependencies)
+          package_json_content = File.read("package.json")
+          manifest = JSON.parse(package_json_content)
```

Comment:
Can we not grab this from `#updated_package_json_content_for_lockfile_name`? It does some memoization, maybe we explicitly don't want that here?

Reply:
Yeah good shout, updating